### PR TITLE
Make server timestamp methods static, rename ServerTimestamps

### PR DESCRIPTION
**Pull request by wilhuff**

### CODE REVIEW

**Comment by wilhuff** [comment]
Another alternative would be to fold these into the `ProtoValues` class.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataWriter.java [hunk 1]
```diff
@@ -90,14 +92,14 @@ Object convertValue(Value value) {
     return result;
   }
 
-  private Object convertServerTimestamp(ServerTimestampValue value) {
+  private Object convertServerTimestamp(Value serverTimestampValue) {
     switch (serverTimestampBehavior) {
       case PREVIOUS:
-        return value.getPreviousValue() == null ? null : convertValue(value.getPreviousValue());
+        Value previousValue = getPreviousValue(serverTimestampValue);
+        if (previousValue == null) return null;
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataWriter.java]: The Java style guide suggests to use braces even for one line statements: https://engdoc.corp.google.com/eng/doc/devguide/java/style/index.md?cl=head#s4.1-braces

File: firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataWriter.java [hunk 2]
```diff
@@ -90,14 +92,14 @@ Object convertValue(Value value) {
...
+        return convertValue(previousValue);
       case ESTIMATE:
-        return !timestampsInSnapshots
-            ? value.getLocalWriteTime().toDate()
-            : value.getLocalWriteTime();
+        return convertTimestamp(getLocalWriteTime(serverTimestampValue));
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataWriter.java]: :) Much better.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataWriter.java (hunk 1 above)
Comment (wilhuff) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/UserDataWriter.java]: Fixed. It's lame that Java and C++ don't align on this point.

**Pull request merged**